components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/processor/SPInitSSOAuthnRequestProcessor.java

```diff
@@ -119,9 +121,11 @@ public boolean canHandle(IdentityRequest identityRequest) {
     @Override
     public FrameworkLoginResponse.FrameworkLoginResponseBuilder process(IdentityRequest identityRequest) throws
             FrameworkException {
+
         CASMessageContext messageContext = new CASMessageContext((CASSInitRequest) identityRequest, new
                 HashMap<String, String>());
-        this.relyingParty = messageContext.getServiceURL();
+        this.relyingParty = CASSSOUtil.getAcsUrl(messageContext.getServiceURL(),
+                messageContext.getRequest().getTenantDomain());
```

Comment:
Shall we move this as a single statement? 
this.relyingParty = CASSSOUtil.getAcsUrl(messageContext.getServiceURL(), messageContext.getRequest().getTenantDomain());